internal/scorecard/alpha/scorecard.go

```diff
@@ -39,52 +44,46 @@ type Scorecard struct {
 	bundleConfigMapName string
 	Client              kubernetes.Interface
 	SkipCleanup         bool
+	TestRunner          TestRunner
+}
+
+type PodTestRunner struct {
+	TestConfig Scorecard
+}
+
+type FakePodTestRunner struct {
+	TestConfig Scorecard
+	TestResult *v1alpha2.ScorecardTestResult
+	Error      error
 }
 
 // RunTests executes the scorecard tests as configured
-func (o Scorecard) RunTests() (testOutput v1alpha2.ScorecardOutput, err error) {
+func (o Scorecard) RunTests(ctx context.Context) (testOutput v1alpha2.ScorecardOutput, err error) {
 	tests := o.selectTests()
 	if len(tests) == 0 {
 		testOutput.Results = make([]v1alpha2.ScorecardTestResult, 0)
 		return testOutput, err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), o.WaitTime)
-	defer cancel()
-
-	bundleData, err := o.getBundleData()
-	if err != nil {
-		return testOutput, fmt.Errorf("error getting bundle data %w", err)
-	}
-
-	err = o.CreateConfigMap(ctx, bundleData)
-	if err != nil {
-		return testOutput, fmt.Errorf("error creating ConfigMap %w", err)
-	}
-
 	testOutput.Results = make([]v1alpha2.ScorecardTestResult, len(tests))
 
 	for idx, test := range tests {
-		result, err := o.runTest(ctx, test)
+		result, err := o.TestRunner.RunTest(ctx, test)
 		if err != nil {
 			result = convertErrorToResult(test.Name, test.Description, err)
 		}
-		testOutput.Results[idx] = result
+		testOutput.Results[idx] = *result
 	}
 
 	if !o.SkipCleanup {
-		defer func() {
-			err := o.deletePods(ctx)
-			if err != nil {
-				testOutput.Results = append(testOutput.Results,
-					convertErrorToResult("", "", err))
-			}
-			err = o.deleteConfigMap(ctx)
-			if err != nil {
-				testOutput.Results = append(testOutput.Results,
-					convertErrorToResult("", "", err))
-			}
-		}()
+		err := o.deletePods(ctx)
+		if err != nil {
+			return testOutput, err
+		}
+		err = o.deleteConfigMap(ctx)
+		if err != nil {
+			return testOutput, err
+		}
```

Comment:
Ditto previous comment. This cleanup is relevant only for the `PodTestRunner`, so we should add a `Cleanup` abstraction to the `TestRunner` interface and use that.

Reply:
cleanup added, called in RunTests() now.